### PR TITLE
Swap BGE-base for Contriever in onboarding docs

### PR DESCRIPTION
**Pull request by xpbowler**
Retrieval results from interactive python code does not exactly match up with results from batch run:

Interactive python code:
```python
from pyserini.search.faiss import FaissSearcher, AutoQueryEncoder

encoder = AutoQueryEncoder('BAAI/bge-base-en-v1.5', device='cpu', pooling='mean',l2_norm=True)
searcher = FaissSearcher('indexes/nfcorpus.bge-base-en-v1.5', encoder)
hits = searcher.search('How to Help Prevent Abdominal Aortic Aneurysms')

for i in range(0, 10):
    print(f'{i+1:2} {hits[i].docid:7} {hits[i].score:.6f}')
```
```
 1 MED-4555 0.791379
 2 MED-4560 0.710725
 3 MED-4421 0.688938
 4 MED-4993 0.686238
 5 MED-4424 0.686214
 6 MED-1663 0.682199
 7 MED-3436 0.680585
 8 MED-2750 0.677033
 9 MED-4324 0.675772
10 MED-2939 0.674646
```

Batch run:
```bash
python -m pyserini.search.faiss \
  --encoder-class auto --encoder BAAI/bge-base-en-v1.5 --l2-norm \
  --index indexes/nfcorpus.bge-base-en-v1.5 \
  --topics collections/nfcorpus/queries.tsv \
  --output runs/run.beir.bge-base-en-v1.5.nfcorpus.txt \
  --batch 128 --threads 32 \
  --hits 1000
```
```bash
$ grep PLAIN-3074 runs/run.beir.bge-base-en-v1.5.nfcorpus.txt | head -10
PLAIN-3074 Q0 MED-4555 1 0.766195 Faiss
PLAIN-3074 Q0 MED-4560 2 0.677510 Faiss
PLAIN-3074 Q0 MED-3436 3 0.656377 Faiss
PLAIN-3074 Q0 MED-1663 4 0.654107 Faiss
PLAIN-3074 Q0 MED-4424 5 0.652038 Faiss
PLAIN-3074 Q0 MED-4421 6 0.651435 Faiss
PLAIN-3074 Q0 MED-4993 7 0.646209 Faiss
PLAIN-3074 Q0 MED-4324 8 0.642927 Faiss
PLAIN-3074 Q0 MED-2939 9 0.642839 Faiss
PLAIN-3074 Q0 MED-2750 10 0.640500 Faiss
```

### CODE REVIEW

**Comment by lintool** [comment]
hey @MXueguang can you take a look at this issue?

**Review by MXueguang — DISMISSED** [review]
lgtm

**Comment by MXueguang** [comment]
this is just for nfcorpus right？ @lintool

**Comment by lintool** [comment]
@MXueguang wait, before merging, I want to sort out the issues... interactive doesn't line up with batch results. Any idea what's going on?

File: docs/experiments-nfcorpus.md [hunk 1]
```diff
@@ -154,8 +154,8 @@ Here's the snippet of Python code that does what we want:
 ```python
 from pyserini.search.faiss import FaissSearcher, AutoQueryEncoder
 
-encoder = AutoQueryEncoder('facebook/contriever-msmarco', device='cpu', pooling='mean')
-searcher = FaissSearcher('indexes/faiss.nfcorpus.contriever-msmacro', encoder)
+encoder = AutoQueryEncoder('BAAI/bge-base-en-v1.5', device='cpu', pooling='mean',l2_norm=True)
```
Comment (lintool) [inline docs/experiments-nfcorpus.md:157]:
```suggestion
encoder = AutoQueryEncoder('BAAI/bge-base-en-v1.5', device='cpu', pooling='mean', l2_norm=True)
```

**Comment by MXueguang** [comment]
@xpbowler sorry I just saw the comment above. One thing is bge take query prefix. Let me check re batch and interactive

**Comment by lintool** [comment]
@MXueguang can you undo 👍 so this doesn't accidentally get merged? thanks!

**Comment by lintool** [comment]
Also, @xpbowler - I was thinking - instead of replace the old contriever model, what if we wrapped it in another collapsible block:

```
<details>
<summary>Try it on another model!</summary>

// move old contriever stuff here?

</details>
```

**Comment by xpbowler** [comment]
@lintool I fixed the bug. I needed to add pooling `mean` in batch run, since it is True in interactive. 

Do we still want to move the old contriever stuff into a collapsible block?

**Comment by xpbowler** [comment]
Ok, sounds good

**Comment by lintool** [comment]
This needs to be updated as well? https://github.com/castorini/pyserini/blob/master/docs/conceptual-framework2.md

**Comment by xpbowler** [comment]
@lintool, updated https://github.com/castorini/pyserini/blob/master/docs/conceptual-framework2.md and added option for using Contriever as well.

File: docs/conceptual-framework2.md [hunk 2]
```diff
@@ -182,6 +182,20 @@ This sorting operation corresponds to top-_k_ retrieval.
 We can see that the output is the same as search with `FaissSearcher` above.
 This is exactly as expected.
 
+<details>
+<summary>Trying it with Contriever:</summary>
+<br/>
+
+To repeat with Contriever, replace:
```
Comment (lintool) [inline docs/conceptual-framework2.md:189]: Can we just have the entire snippet of code instead of replacing _X_ with _Y_?

Also, report expected response?

**Review by lintool — CHANGES_REQUESTED** [review]

File: docs/conceptual-framework2.md (hunk 2 above)
Comment (xpbowler) [reply docs/conceptual-framework2.md]: done

File: docs/experiments-nfcorpus.md [hunk 3]
```diff
@@ -77,6 +77,25 @@ Okay, the data are ready now.
...
+  output  --embeddings indexes/nfcorpus.bge-base-en-v1.5 \
+          --to-faiss \
+  encoder --encoder BAAI/bge-base-en-v1.5 --l2-norm \
+          --device cpu \
+          --pooling mean \
+          --fields title text \
+          --batch 32
+```
+
+We're using the [`BAAI/bge-base-en-v1.5](https://huggingface.co/BAAI/bge-base-en-v1.5) encoder, which can be found on HuggingFace.
```
Comment (lintool) [inline docs/experiments-nfcorpus.md:93]:
```suggestion
We're using the [`BAAI/bge-base-en-v1.5`](https://huggingface.co/BAAI/bge-base-en-v1.5) encoder, which can be found on HuggingFace.
```

**Review by lintool — APPROVED** [review]

**Pull request merged**